tfhe/src/typed_api/booleans/client_key.rs

```diff
@@ -0,0 +1,25 @@
+use crate::boolean::client_key::ClientKey;
+
+use serde::{Deserialize, Serialize};
+
+use super::{parameters::BooleanParameterSet, FheBoolParameters, types::r#static::StaticBoolParameters};
+
+ #[cfg_attr(doc, cfg(feature = "boolean"))]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct GenericBoolClientKey<P>
+where
+    P: BooleanParameterSet,
+{
+    pub(in crate::typed_api::booleans) key: ClientKey,
+    _marker: std::marker::PhantomData<P>,
+}
+
+impl From<FheBoolParameters> for GenericBoolClientKey<StaticBoolParameters>
```

Comment:
I'll probably ask "dumb" questions about FheBoolParameters and StaticBoolParameters and so on, it's not yet clear to me how all these things work together

Is there somewhere where an overview of how the types interact is written ?